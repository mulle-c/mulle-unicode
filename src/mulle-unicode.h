//
//  mulle-unicode.h
//  mulle-unicode
//
//  Created by Nat! on 18.03.16.
//  Copyright Â© 2016 Mulle kybernetiK.
//  Copyright (c) 2016 Codeon GmbH.
//  All rights reserved.
//
//
//  Redistribution and use in source and binary forms, with or without
//  modification, are permitted provided that the following conditions are met:
//
//  Redistributions of source code must retain the above copyright notice, this
//  list of conditions and the following disclaimer.
//
//  Redistributions in binary form must reproduce the above copyright notice,
//  this list of conditions and the following disclaimer in the documentation
//  and/or other materials provided with the distribution.
//
//  Neither the name of Mulle kybernetiK nor the names of its contributors
//  may be used to endorse or promote products derived from this software
//  without specific prior written permission.
//
//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
//  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
//  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
//  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
//  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
//  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
//  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
//  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
//  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
//  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
//  POSSIBILITY OF SUCH DAMAGE.
//
#ifndef mulle_unicode_h__
#define mulle_unicode_h__

#define MULLE_UNICODE_VERSION  ((2 << 20) | (4 << 8) | 6)

#include "include.h"

#include <stdint.h>

// #include "_mulle-unicode-provide.h"

// manual inclusion of headers, because we don't want to expose all the
// .inc files

#include "mulle-unicode-ctype.h"
#include "mulle-unicode-is-alphanumeric.h"
#include "mulle-unicode-is-capitalized.h"
#include "mulle-unicode-is-control.h"
#include "mulle-unicode-is-decimaldigit.h"
#include "mulle-unicode-is-decomposable.h"
#include "mulle-unicode-is-legalcharacter.h"
#include "mulle-unicode-is-letter.h"
#include "mulle-unicode-is-lowercase.h"
#include "mulle-unicode-is-newline.h"
#include "mulle-unicode-is-nonbase.h"
#include "mulle-unicode-is-noncharacter.h"
#include "mulle-unicode-is-punctuation.h"
#include "mulle-unicode-is-symbol.h"
#include "mulle-unicode-is-uppercase.h"
#include "mulle-unicode-is-whitespace.h"
#include "mulle-unicode-is-whitespaceornewline.h"
#include "mulle-unicode-is-zerodigit.h"
#include "mulle-unicode-tolower.h"
#include "mulle-unicode-toupper.h"

#ifdef __has_include
# if __has_include( "_mulle-unicode-versioncheck.h")
#  include "_mulle-unicode-versioncheck.h"
# endif
#endif


#endif /* mulle_unicode_h */
