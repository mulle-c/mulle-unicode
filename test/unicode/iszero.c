#include <mulle-unicode/mulle-unicode.h>
#include <stdio.h>
#include <stdlib.h>


int  main()
{
   int32_t  c;

   for( c = 0; c < 0x110000; c++)
   {
      if( mulle_unicode_is_zerodigit( c))
         printf( "%#0x is zero\n", c);
   }
   return( 0);
}

