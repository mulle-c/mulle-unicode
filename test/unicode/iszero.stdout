0x30 is zero
0x660 is zero
0x6f0 is zero
0x7c0 is zero
0x966 is zero
0x9e6 is zero
0xa66 is zero
0xae6 is zero
0xb66 is zero
0xbe6 is zero
0xc66 is zero
0xc78 is zero
0xce6 is zero
0xd66 is zero
0xde6 is zero
0xe50 is zero
0xed0 is zero
0xf20 is zero
0x1040 is zero
0x1090 is zero
0x17e0 is zero
0x1810 is zero
0x1946 is zero
0x19d0 is zero
0x1a80 is zero
0x1a90 is zero
0x1b50 is zero
0x1bb0 is zero
0x1c40 is zero
0x1c50 is zero
0x24ea is zero
0x24ff is zero
0xa620 is zero
0xa8d0 is zero
0xa8e0 is zero
0xa900 is zero
0xa9d0 is zero
0xa9f0 is zero
0xaa50 is zero
0xabf0 is zero
0xff10 is zero
0x104a0 is zero
0x10d30 is zero
0x11066 is zero
0x110f0 is zero
0x11136 is zero
0x111d0 is zero
0x112f0 is zero
0x11366 is zero
0x11450 is zero
0x114d0 is zero
0x11650 is zero
0x116c0 is zero
0x11730 is zero
0x118e0 is zero
0x11c50 is zero
0x11d50 is zero
0x11da0 is zero
0x16a60 is zero
0x16b50 is zero
0x16e80 is zero
0x1d7ce is zero
0x1d7d8 is zero
0x1d7e2 is zero
0x1d7ec is zero
0x1d7f6 is zero
0x1e140 is zero
0x1e2f0 is zero
0x1e950 is zero
0x1f100 is zero
0x1f101 is zero
0x1f10b is zero
0x1f10c is zero
0xe0030 is zero
